command/deploy.go

```diff
@@ -38,53 +38,41 @@ func defaultDeployOutputOptions() *deployOutputOptions {
 
 var testFailureError = errors.New("Apex tests failed")
 
-func monitorDeploy(deployId string) (ForceCheckDeploymentStatusResult, error) {
-	var result ForceCheckDeploymentStatusResult
-	var err error
-	retrying := false
-	for {
-		result, err = force.Metadata.CheckDeployStatus(deployId)
-		if err != nil {
-			if retrying {
-				return result, fmt.Errorf("Error getting deploy status: %w", err)
-			} else {
-				retrying = true
-				Log.Info(fmt.Sprintf("Received error checking deploy status: %s.  Will retry once before aborting.", err.Error()))
-			}
-		} else {
-			retrying = false
-		}
-		if result.Done {
-			break
-		}
-		if !retrying {
-			Log.Info(result)
-		}
-		time.Sleep(5000 * time.Millisecond)
-	}
-	return result, err
+type deployStatus struct {
+	mu      sync.Mutex
+	aborted bool
+}
+
+func (c *deployStatus) abort() {
+	c.mu.Lock()
+	c.aborted = true
+	c.mu.Unlock()
+}
+
+func (c *deployStatus) isAborted() bool {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	return c.aborted
 }
 
 func deploy(force *Force, files ForceMetadataFiles, deployOptions *ForceDeployOptions, outputOptions *deployOutputOptions) error {
-	if outputOptions.quiet {
-		previousLogger := Log
-		var l quietLogger
-		Log = l
-		defer func() {
-			Log = previousLogger
-		}()
-	}
+	status := deployStatus{aborted: false}
+
+	return deployWith(force, &status, files, deployOptions, outputOptions)
+}
+
+func deployWith(force *Force, status *deployStatus, files ForceMetadataFiles, deployOptions *ForceDeployOptions, outputOptions *deployOutputOptions) error {
 	startTime := time.Now()
 	deployId, err := force.Metadata.StartDeploy(files, *deployOptions)
 	if err != nil {
-		ErrorAndExit(err.Error())
+		return err
 	}
 	stopDeployUponSignal(force, deployId)
 	if outputOptions.interactive {
```

Comment:
We'll need to make `--interactive` with multiple accounts an error, at least until we explicitly support it.